resim/metrics/python/metrics.py

```diff
@@ -27,8 +40,16 @@
 
 MetricT = TypeVar('MetricT', bound='Metric')
 
+def metric_dataclass(*args: Callable) -> Callable:
+    return dataclasses.dataclass(
+        *args,
+        init=False,
+        kw_only=True,
+        repr=True,
+        eq=False
```

Comment:
The automatically-generated __eq__ methods were overriding the base class's __eq__ so I turned this off.

Reply:
(Interesting - I mostly made my methods in order to avoid potential expensive recursions, but I'm actually not that averse to leaving the default if you want.)

Reply:
If I'm understanding the comment correctly, I think this still avoids recursing through dependent data. The problem was that the concrete classes would have their own equality operators generated which would override the base class's. When you turn this off, the subclasses have no option but to use the id-comparison approach that the base class __eq__() method employs.